src/RoomNotifs.js

```diff
@@ -35,6 +35,27 @@ function _shouldShowMentionBadge(roomNotifState) {
     return roomNotifState !== MUTE;
 }
 
+export function countRoomsWithNotif(rooms) {
+    return rooms.reduce((result, room, index) => {
+        const roomNotifState = getRoomNotifsState(room.roomId);
+        const highlight = room.getUnreadNotificationCount('highlight') > 0;
+        const notificationCount = room.getUnreadNotificationCount();
+
+        const notifBadges = notificationCount > 0 && _shouldShowNotifBadge(roomNotifState);
+        const mentionBadges = highlight && _shouldShowMentionBadge(roomNotifState);
+        const isInvite = room.hasMembershipState(MatrixClientPeg.get().credentials.userId, 'invite');
+        const badges = notifBadges || mentionBadges || isInvite;
+
+        if (badges) {
+            result.count++;
+            if (highlight) {
+                result.highlight = true;
```

Comment:
Are you computing `highlight` for some future consumer? It's not used below...

Reply:
Yeah, I figured it just ought to be consistent with the other function below.